### PR TITLE
[Merged by Bors] - Improve docs and naming for RawWindowHandle functionality

### PR DESCRIPTION
**Pull request by alice-i-cecile**
# Objective

- As noticed in #4333 by @x-52, the exact purpose and logic of `HasRawWIndowHandleWrapper` is unclear
- Unfortunately, there are rather good reasons why this design is needed (and why we can't just `impl HasRawWindowHandle for RawWindowHandleWrapper` 

## Solution

- Rename `HasRawWindowHandleWrapper` to `ThreadLockedRawWindowHandleWrapper`, reflecting the primary distinction
- Document how this design is intended to be used
- Leave comments explaining why this design must exist

## Migration Guide

- renamed `HasRawWindowHandleWrapper` to `ThreadLockedRawWindowHandleWrapper`

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
Pulling in @cart directly for a review here as the original author of the code. I'm reasonably confident that I understand the logic behind the design now, but this is tricksy unsafe logic.

File: crates/bevy_window/src/raw_window_handle.rs [hunk 1]
```diff
@@ -27,10 +32,22 @@ impl RawWindowHandleWrapper {
 unsafe impl Send for RawWindowHandleWrapper {}
 unsafe impl Sync for RawWindowHandleWrapper {}
 
-pub struct HasRawWindowHandleWrapper(RawWindowHandle);
+/// A [`RawWindowHandleWrapper`] that cannot be sent across threads
```
Comment (nicopap) [inline crates/bevy_window/src/raw_window_handle.rs:35]:
```suggestion
/// A [`RawWindowHandleWrapper`] that cannot be sent across threads.
```

File: crates/bevy_window/src/raw_window_handle.rs [hunk 2]
```diff
@@ -10,12 +13,14 @@ impl RawWindowHandleWrapper {
         Self(handle)
     }
 
+    /// Returns a [`HasRawWindowHandle`] impl, which exposes [`RawWindowHandle`]
```
Comment (nicopap) [inline crates/bevy_window/src/raw_window_handle.rs:16]:
```suggestion
    /// Returns a [`HasRawWindowHandle`] impl, which exposes [`RawWindowHandle`].
```

File: crates/bevy_window/src/raw_window_handle.rs [hunk 3]
```diff
@@ -1,7 +1,10 @@
 use raw_window_handle::{HasRawWindowHandle, RawWindowHandle};
 
-/// This wrapper exist to enable safely passing a [`RawWindowHandle`] across threads. Extracting the handle
-/// is still an unsafe operation, so the caller must still validate that using the raw handle is safe for a given context.
+/// A wrapper over [`RawWindowHandle`] that allows us to safely pass it across threads.
+///
+/// Depending on the platform, the underlying pointer-containing handle cannot be used on all threads,
+/// and so we cannot simply make it (or any type that has a safe opration to get a [`RawWindowHandle`])
```
Comment (nicopap) [inline crates/bevy_window/src/raw_window_handle.rs:6]:
```suggestion
/// and so we cannot simply make it (or any type that has a safe operation to get a [`RawWindowHandle`])
```

**Review by nicopap — CHANGES_REQUESTED** [review]

**Review by nicopap — APPROVED** [review]

**Pull request closed without merging**